ecommerce/api.py

```diff
@@ -107,6 +109,37 @@ def generate_checkout_payload(request):
     return payload
 
 
+def check_discount_for_products(discount, basket):
+    """
+    Checks the validity of the discount against what's in the basket.
+
+    If the discount either has no products associated with it, or the products
+    in the basket are applicable to the discount, this returns True.
+    Otherwise, returns False.
+
+    Args:
+        - basket (Basket): the current basket
+        - discount (Discount|string: the discount to apply (if a string, loads the discount code specified)
+    Returns:
+        boolean
+    """
+    if not isinstance(discount, Discount):
+        discount = Discount.objects.filter(discount_code=discount).first()
+
+    basket_item_products = basket.get_products()
```

Comment:
I think it would make more sense to name this `basket_products`, meaning products in a basket.

Reply:
Done in https://github.com/mitodl/mitxonline/pull/708/commits/fe65eb05b4c6f846d4b157a46804f0f800656674